hazelcast/src/main/java/com/hazelcast/concurrent/atomiclong/operations/MergeOperation.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.concurrent.atomiclong.operations;
+
+import com.hazelcast.concurrent.atomiclong.AtomicLongService;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.spi.Operation;
+import com.hazelcast.spi.SplitBrainMergeEntryView;
+import com.hazelcast.spi.SplitBrainMergePolicy;
+
+import java.io.IOException;
+
+import static com.hazelcast.concurrent.atomiclong.AtomicLongDataSerializerHook.MERGE;
+import static com.hazelcast.spi.merge.SplitBrainEntryViews.createSplitBrainMergeEntryView;
+
+public class MergeOperation extends AtomicLongBackupAwareOperation {
+
+    private SplitBrainMergePolicy mergePolicy;
+    private long mergingValue;
+
+    private transient Long backupValue;
+
+    public MergeOperation() {
+    }
+
+    public MergeOperation(String name, SplitBrainMergePolicy mergePolicy, long mergingValue) {
+        super(name);
+        this.mergePolicy = mergePolicy;
+        this.mergingValue = mergingValue;
+    }
+
+    @Override
+    public void run() throws Exception {
+        AtomicLongService service = getService();
+        boolean isExistingContainer = service.containsAtomicLong(name);
```

Comment:
This is a little trick to find out, if the `AtomicLong` is existing in the larger cluster or not. This is needed to make the `PutIfAbsentMergePolicy` working correctly, since the initial value `0` could also be set by the user. I'm passing this state in the unused `key` field of the `SplitBrainMergeEntryView` and use this information in the `AtomicLongContainer` to determine if an `existingEntry` should be created or not.